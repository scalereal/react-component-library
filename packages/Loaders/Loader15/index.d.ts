import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const Loader15: React.FC<Props> & WithStyle;
export default Loader15;
