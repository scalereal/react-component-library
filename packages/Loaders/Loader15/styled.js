var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    position: relative;\n    width: .15em;\n    height: .15em;\n    background-color: currentcolor;\n    border-radius: 100%;\n    animation: loader-39-1 30s infinite linear;\n\t&:before, &:after {\n\t\tcontent: '';\n\t\tborder-radius: 100%;\n\t\tposition: absolute;\n\t\ttop: 50%;\n\t\tleft: 50%;\n\t\ttransform: translate(-50%, -50%);\n\t}\n\t&:before {\n\t\twidth: .3em;\n\t\theight: 1em;\n\t\tanimation: loader-39-2 .8s linear infinite;\n\t}\n\t&:after {\n\t\twidth: 1em;\n\t\theight: .3em;\n\t\tanimation: loader-39-2 1.2s linear infinite;\n\t}\n\n    @keyframes loader-39-1 {\n        0% {\n            transform: rotate(0deg);\n        }\n        100% {\n            transform: rotate(360deg);\n        }\n    }\n    @keyframes loader-39-2 {\n        0% {\n            box-shadow: 0.04em -0.04em 0 0.02em currentcolor;\n        }\n        25% {\n            box-shadow: 0.04em 0.04em 0 0.02em currentcolor;\n        }\n        50% {\n            box-shadow: -0.04em 0.04em 0 0.02em currentcolor;\n        }\n        75% {\n            box-shadow: -0.04em -0.04em 0 0.02em currentcolor;\n        }\n        100% {\n            box-shadow: 0.04em -0.04em 0 0.02em currentcolor;\n        }\n    }\n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    position: relative;\n    width: .15em;\n    height: .15em;\n    background-color: currentcolor;\n    border-radius: 100%;\n    animation: loader-39-1 30s infinite linear;\n\t&:before, &:after {\n\t\tcontent: '';\n\t\tborder-radius: 100%;\n\t\tposition: absolute;\n\t\ttop: 50%;\n\t\tleft: 50%;\n\t\ttransform: translate(-50%, -50%);\n\t}\n\t&:before {\n\t\twidth: .3em;\n\t\theight: 1em;\n\t\tanimation: loader-39-2 .8s linear infinite;\n\t}\n\t&:after {\n\t\twidth: 1em;\n\t\theight: .3em;\n\t\tanimation: loader-39-2 1.2s linear infinite;\n\t}\n\n    @keyframes loader-39-1 {\n        0% {\n            transform: rotate(0deg);\n        }\n        100% {\n            transform: rotate(360deg);\n        }\n    }\n    @keyframes loader-39-2 {\n        0% {\n            box-shadow: 0.04em -0.04em 0 0.02em currentcolor;\n        }\n        25% {\n            box-shadow: 0.04em 0.04em 0 0.02em currentcolor;\n        }\n        50% {\n            box-shadow: -0.04em 0.04em 0 0.02em currentcolor;\n        }\n        75% {\n            box-shadow: -0.04em -0.04em 0 0.02em currentcolor;\n        }\n        100% {\n            box-shadow: 0.04em -0.04em 0 0.02em currentcolor;\n        }\n    }\n"])), function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n            " + Loader + "{\n                margin: 10px;\n            }\n           ";
        case "XS":
            return "font-size: 1rem;\n            " + Loader + "{\n                margin: 10px;\n            }\n            \n            ";
        case "S":
            return "font-size: 2rem;\n            " + Loader + "{\n                margin: 17px;\n            }\n            ";
        case "M":
            return "font-size: 3rem;\n            " + Loader + "{\n                margin: 25px;\n            }\n            ";
        case "L":
            return "font-size: 5rem;\n            " + Loader + "{\n                margin: 45px;\n            }\n            ";
        case "XL":
            return "font-size: 6rem;\n            " + Loader + "{\n                margin: 55px;\n            }\n            ";
        default:
            return "font-size: 1rem;\n        " + Loader + "{\n            margin: 10px;\n        }\n       ";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.visible ? 'inline-block' : 'none'; });
var templateObject_1, templateObject_2;
