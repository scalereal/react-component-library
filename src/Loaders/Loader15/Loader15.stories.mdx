import Loader15 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader15"
    component={Loader15}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader15 {...args} />
    </div>
);

<Preview>
    <Story name="Loader15" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader15 size="S" Color="green" />
        <Loader15 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader15 size="XXS" />
        <Loader15 size="XS" />
        <Loader15 size="S" />
        <Loader15 size="M" />
        <Loader15 size="L" />
        <Loader15 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader15" />
