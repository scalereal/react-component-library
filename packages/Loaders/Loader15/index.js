var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { Loader, LoaderWrapper } from './styled';
var Loader15 = React.memo(React.forwardRef(function (props, ref) {
    var size = props.size;
    return (React.createElement(LoaderWrapper, __assign({ ref: ref, size: size }, props),
        React.createElement(Loader, __assign({}, props))));
}));
Loader15.displayName = "Loader15";
Loader15.defaultProps = {
    size: 'S',
    visible: true
};
export default Loader15;
