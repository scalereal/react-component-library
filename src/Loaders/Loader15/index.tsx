import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const Loader15:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

Loader15.displayName="Loader15";
Loader15.defaultProps = {
    size:'S',
    visible: true
};

export default Loader15;
