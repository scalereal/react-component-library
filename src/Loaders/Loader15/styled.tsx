import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    position: relative;
    width: .15em;
    height: .15em;
    background-color: currentcolor;
    border-radius: 100%;
    animation: loader-39-1 30s infinite linear;
	&:before, &:after {
		content: '';
		border-radius: 100%;
		position: absolute;
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
	}
	&:before {
		width: .3em;
		height: 1em;
		animation: loader-39-2 .8s linear infinite;
	}
	&:after {
		width: 1em;
		height: .3em;
		animation: loader-39-2 1.2s linear infinite;
	}

    @keyframes loader-39-1 {
        0% {
            transform: rotate(0deg);
        }
        100% {
            transform: rotate(360deg);
        }
    }
    @keyframes loader-39-2 {
        0% {
            box-shadow: 0.04em -0.04em 0 0.02em currentcolor;
        }
        25% {
            box-shadow: 0.04em 0.04em 0 0.02em currentcolor;
        }
        50% {
            box-shadow: -0.04em 0.04em 0 0.02em currentcolor;
        }
        75% {
            box-shadow: -0.04em -0.04em 0 0.02em currentcolor;
        }
        100% {
            box-shadow: 0.04em -0.04em 0 0.02em currentcolor;
        }
    }
`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                margin: 10px;
            }
           `;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                margin: 10px;
            }
            
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                margin: 17px;
            }
            `;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                margin: 25px;
            }
            `;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                margin: 45px;
            }
            `;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                margin: 55px;
            }
            `;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            margin: 10px;
        }
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;
