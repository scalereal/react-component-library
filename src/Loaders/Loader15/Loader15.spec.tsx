import { render } from '@testing-library/react';
import * as React from 'react';
import Loader15 from '.';

describe('Loader15 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader15/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader15
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader15 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});