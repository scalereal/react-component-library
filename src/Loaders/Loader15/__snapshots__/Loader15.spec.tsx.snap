// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader15 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz eelQNL"
  >
    <div
      class="sc-bdfBwQ hKKOmu"
    />
  </div>
</div>
`;

exports[`Loader15 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz DBHwP"
  >
    <div
      class="sc-bdfBwQ fSqboP"
    />
  </div>
</div>
`;

exports[`Loader15 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz iqVWtv"
  >
    <div
      class="sc-bdfBwQ hKKOmu"
    />
  </div>
</div>
`;
